core/dictgen/src/rootcling_impl.cxx

```diff
@@ -772,9 +772,17 @@ string gLibsNeeded;
 
 ////////////////////////////////////////////////////////////////////////////////
 
-void RecordDeclCallback(const char *c)
+void RecordDeclCallback(const clang::RecordDecl* recordDecl)
 {
-   string need(gAutoloads[c]);
+   std::string need;
+   if (recordDecl->hasOwningModule()) {
+      clang::Module *M = recordDecl->getOwningModule()->getTopLevelModule();
+      need = "lib" + M->Name + ".so";
```

Comment:
You might not need the .so (and some platform it would actually be .dll).
Are you sure you *always* need to add the 'lib' prefix?